src/app/components/Icons/CircleIconWithChildren.tsx

```diff
@@ -1,19 +1,17 @@
-import * as React from 'react';
+import React from 'react';
 import './Icons.css';
 
-export interface CircleIconWithTextProps {
+export interface CircleIconWithChildrenProps {
     backgroundColor?: string;
-    text?: string;
     borderColor?: string;
     color?: string;
```

Comment:
I've taken this approach in the past but found that if you expose the style property you can make things more general but not lose out on having sensible defaults by spreading the style property overtop of your defaults.  That way if you want to be able to change a different style related property you don't have to expose a new property in your props.

```ts
export interface CircleIconProps{
   style?:React.CSSProperties;
}

class CircleIcon{

render(){
  const {style = {}} = this.props;
return (
             <div 
                 className="circle-icon" 
                 style={{
                     fontSize: 12, 
                     borderColor:'grey', 
                     color:'grey', 
                     backgroundColor:'white', 
                     fontWeight: 'normal',
                     ...style
                 }}
             >
                {this.props.children}
             </div>
         );   

}

}

```